source/DSCResources/DSC_CMCollections/DSC_CMCollections.psm1

```diff
@@ -0,0 +1,685 @@
+$script:dscResourceCommonPath = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\DscResource.Common'
+$script:configMgrResourcehelper = Join-Path -Path $PSScriptRoot -ChildPath '..\..\Modules\ConfigMgrCBDsc.ResourceHelper'
+
+Import-Module -Name $script:dscResourceCommonPath
+Import-Module -Name $script:configMgrResourcehelper
+
+$script:localizedData = Get-LocalizedData -DefaultUICulture 'en-US'
+
+<#
+    .SYNOPSIS
+        This will return a hashtable of results.
+
+    .PARAMETER SiteCode
+        Specifies the site code for Configuration Manager site.
+
+    .PARAMETER CollectionName
+        Specifies a name for the collection.
+
+    .PARAMETER CollectionType
+        Specifies the type of collection. Valid values are User and Device.
+
+    .PARAMETER LimitingCollectionName
+        Specifies the name of a collection to use as the default scope for this collection.
+        Limiting collection is not evaluated in Test and is only used
+        if the collection needs created.
+
+    .PARAMETER Comment
+        Specifies a comment for the collection.
+
+    .PARAMETER RefreshSchedule
+        Specifies a schedule that determines when Configuration Manager refreshes the collection.
+
+    .PARAMETER RefreshType
+        Specifies how Configuration Manager refreshes the collection.
+        Valid values are: Manual, Periodic, Continuous, and Both.
+
+    .PARAMETER QueryRules
+        Specifies the name of the Rule and the query expression that Configuration Manager uses to update collections.
+
+    .PARAMETER ExcludeMembership
+        Specifies the collection name to exclude members from. If clients are in the excluded collection they will
+        not be added to the collection.
+
+    .PARAMETER DirectMembership
+        Specifies the resourceid for the direct membership rule.
+
+    .PARAMETER Ensure
+        Specifies if the collection is to be present or absent.
+#>
+function Get-TargetResource
```

Comment:
There are non mandatory parameters that are being used in Get.

Reply:
Updating.  Need to change Pester Tests to account for this and also to add Ensure param to return.